app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala

```diff
@@ -285,7 +292,21 @@ class BusinessEntityController @Inject()(
           form => {
             form.value match {
               case "true" =>
-                escService.removeMember(undertakingRef, removeBE).map(_ => Redirect(routes.BusinessEntityController.getAddBusinessEntity()))
+                val removalEffectiveDateString = DateFormatter.govDisplayFormat(timeProvider.today)
+                for {
+                  _ <- escService.removeMember(undertakingRef, removeBE)
+                  emailAddressBE <- retrieveEmailService.retrieveEmailByEORI(removeBE.businessEntityIdentifier).map(_.getOrElse(handleMissingSessionData("Business entity Email")))
+                  emailAddressLead <- retrieveEmailService.retrieveEmailByEORI(eori).map(_.getOrElse(handleMissingSessionData("Lead EORI Email Address")))
+                  templateIdBE = TemplateHelpers.getTemplateId(configuration, removeMemberEmailToBusinessEntity)
+                  templateIdLead = TemplateHelpers.getTemplateId(configuration, removeMemberEmailToLead)
+                  emailParametersBE = SingleEORIAndDateEmailParameter(removeBE.businessEntityIdentifier, undertaking.name, undertakingRef, removalEffectiveDateString,  "Email to BE for being removed as a member")
+                  emailParametersLead = DoubleEORIAndDateEmailParameter(eori, removeBE.businessEntityIdentifier,  undertaking.name, undertakingRef, removalEffectiveDateString, "Email to Lead  for removing a new member")
+                } yield {
+                  sendEmailService.sendEmail(emailAddressBE, emailParametersBE, templateIdBE)
```

Comment:
What happens if either send email request fails? Is it possible to batch up two different email requests so we can do this in one request rather than two?

Reply:
If email send fails , then there should be no error. user should be able to use the application. There is no api call which will batch up the request.

Reply:
Besides each email is different. The email to lead eori is different than the the email send to business entity and hence have separate templates.